File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionPaymentRepository.java
package com.eager.questioncloud.core.domain.payment.infrastructure.repository;

import com.eager.questioncloud.core.domain.payment.model.QuestionPayment;

public interface QuestionPaymentRepository {
    QuestionPayment save(QuestionPayment questionPayment);

    int countByUserId(Long userId);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionPaymentRepository.kt
package com.eager.questioncloud.core.domain.payment.infrastructure.repository

import com.eager.questioncloud.core.domain.payment.model.QuestionPayment

interface QuestionPaymentRepository {
    fun save(questionPayment: QuestionPayment): QuestionPayment

    fun countByUserId(userId: Long): Int
}
